packages/@expo/cli/src/utils/codesigning.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  convertCertificatePEMToCertificate,
+  convertKeyPairToPEM,
+  convertCSRToCSRPEM,
+  generateKeyPair,
+  generateCSR,
+  convertPrivateKeyPEMToPrivateKey,
+  validateSelfSignedCertificate,
+  signStringRSASHA256AndVerify,
+} from '@expo/code-signing-certificates';
+import { ExpoConfig } from '@expo/config';
+import { promises as fs } from 'fs';
+import { pki as PKI } from 'node-forge';
+import path from 'path';
+import { Dictionary, parseDictionary } from 'structured-headers';
+
+import { getExpoGoIntermediateCertificateAsync } from '../api/getExpoGoIntermediateCertificate';
+import { getProjectDevelopmentCertificateAsync } from '../api/getProjectDevelopmentCertificate';
+import { APISettings } from '../api/settings';
+import * as Log from '../log';
+import { createTemporaryProjectFile } from '../start/project/dotExpo';
+
+export type CodeSigningInfo = {
+  privateKey: string;
+  certificateForPrivateKey: string;
+  certificateChainForResponse: string[]; // empty means no need to serve the certificate chain in the multipart response
```

Comment:
Helpful to document that these chain fields should be ordered starting with the leaf signing certificates